execution/model/operator.go

```diff
@@ -14,6 +15,44 @@ import (
 	"github.com/thanos-io/promql-engine/query"
 )
 
+type metadataKey string
```

Comment:
we have similar pattern in /execution/warning  i think, should we move this to /execution/metadata so its similarly structured?